From fe8d39dc2b437ba4920161b66e221ae73c976bea Mon Sep 17 00:00:00 2001
From: Jonathan Bell <jonathan@raspberrypi.com>
Date: Fri, 9 Feb 2024 13:32:38 +0000
Subject: [PATCH 0890/1085] dts: bcm2712: update sdio1 on Pi 5

Switch to using card-detection via GPIO, and add missing emmc_cmd pin.
Also, "emmc_*" isn't the name of the respective function, but the name
of the pin. These pins are single-function, but need pulls set
accordingly.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
---
 arch/arm/boot/dts/broadcom/bcm2712-rpi-5-b.dts | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/arch/arm/boot/dts/broadcom/bcm2712-rpi-5-b.dts
+++ b/arch/arm/boot/dts/broadcom/bcm2712-rpi-5-b.dts
@@ -359,7 +359,7 @@ dpi_16bit_gpio2:        &rp1_dpi_16bit_g
 	sd-uhs-sdr50;
 	sd-uhs-ddr50;
 	sd-uhs-sdr104;
-	//broken-cd;
+	cd-gpios = <&gio_aon 5 GPIO_ACTIVE_LOW>;
 	//no-1-8-v;
 	status = "okay";
 };
@@ -396,7 +396,7 @@ dpi_16bit_gpio2:        &rp1_dpi_16bit_g
 	};
 
 	emmc_sd_pulls: emmc_sd_pulls {
-		function = "emmc_dat0", "emmc_dat1", "emmc_dat2", "emmc_dat3";
+		pins = "emmc_cmd", "emmc_dat0", "emmc_dat1", "emmc_dat2", "emmc_dat3";
 		bias-pull-up;
 	};
 };
